SPECS/coredns/coredns.spec

```diff
@@ -31,18 +31,21 @@ Source0:        %{name}-%{version}.tar.gz
 #       - For the value of "--mtime" use the date "2021-04-26 00:00Z" to simplify future updates.
 Source1:        %{name}-%{version}-vendor.tar.gz
 Patch0:         makefile-buildoption-commitnb.patch
+Patch1:         CVE-2023-49295.patch
 
 BuildRequires:  golang >= 1.12
 
 %description
 CoreDNS is a fast and flexible DNS server.
 
 %prep
-%autosetup -p1
-
-%build
+%setup -q
+%patch0 -p1
```

Comment:
My suggestion would be to leverage a combo os `%autosetup & %autopatch`.

Something like this:
```
%autosetup -N
%autopatch -m 1000
```

We rename the vendor patch from `Patch1` -> `Patch1001` and apply on line 46
`%patch1001 -p1`

tl;dr

```
%autosetup -N -> -N disables automatic patch application if necessary for some reason. If %autosetup is called with -N, the patch-application phase can be manually invoked with %autopatch macro.

%autopatch -M 1000 -> -M<number> Apply patches up to <number> range
```

Reference: https://rpm-software-management.github.io/rpm/manual/autosetup.html

Reply:
Applied the suggestion and re-run the buddy build.